frontend/lib/src/components/elements/GraphVizChart/GraphVizChart.tsx

```diff
@@ -38,75 +38,68 @@ interface Dimensions {
 const dummyGraphviz = graphviz
 dummyGraphviz // eslint-disable-line @typescript-eslint/no-unused-expressions
 
+const isFullScreen = (height = 0): boolean => Boolean(height)
+
 export function GraphVizChart({
   width: propWidth,
   element,
   height: propHeight,
 }: GraphVizChartProps): ReactElement {
   const chartId = `graphviz-chart-${element.elementId}`
 
+  const isFull = isFullScreen(propHeight)
+  const isFullRef = useRef(isFull)
+  // Update isFull state whenever propHeight changes
+  useEffect(() => {
+    isFullRef.current = isFull
+  }, [isFull])
+
   let originalHeight = 0
   let originalWidth = 0
 
-  const getChartData = (): string => {
-    return element.spec
-  }
+  const setSvgDimensions = useMemo(() => {
+    return (node: SVGGraphicsElement): void => {
+      const bbox = node.getBBox()
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+      originalHeight = Math.round(bbox.height)
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+      originalWidth = Math.round(bbox.width)
 
-  const getChartDimensions = (): Dimensions => {
-    let chartWidth = originalWidth
-    let chartHeight = originalHeight
-
-    if (propHeight) {
-      // fullscreen
-      chartWidth = propWidth
-      chartHeight = propHeight
-    } else if (element.useContainerWidth) {
-      chartWidth = propWidth
+      select(node)
+        .attr("width", isFullRef.current ? "100%" : `${originalWidth}pt`)
+        .attr("height", isFullRef.current ? "100%" : `${originalHeight}pt`)
     }
+  }, [isFullRef])
```

Comment:
Nit: instead of twice above, should be able to put the 
`// eslint-disable-next-line react-hooks/exhaustive-deps` right above the dependency array